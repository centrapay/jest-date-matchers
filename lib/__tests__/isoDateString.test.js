/*
 * Copyright 2020 Centrapay
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

'use strict';

require('../../');

describe('toBeIsoDateStringNear', () => {

  describe('default threshold', () => {

    describe('affirmative', () => {

      const examples = [
        { value: '2020-09-01T10:44:59.000Z', target: '2020-09-01T10:45:00.000Z' },
        { value: '2020-09-01T10:45:00.000Z', target: '2020-09-01T10:45:00.000Z' },
        { value: '2020-09-01T10:45:01.000Z', target: '2020-09-01T10:45:00.000Z' }
      ];

      examples.forEach(({ value, target }) => {
        test(`${value} is near ${target}`, () => {
          expect(value).toBeIsoDateStringNear(target);
        });
      });

      examples.forEach(({ value, target }) => {
        test(`failure message: ${value} near ${target}`, () => {
          try {
            expect(value).not.toBeIsoDateStringNear(target);
          } catch(e) {
            expect(e.message).toMatchSnapshot();
          }
        });
      });

    });

    describe('negative', () => {

      const examples = [
        { value: '2020-09-01T10:44:58.000Z', target: '2020-09-01T10:45:00.000Z' },
        { value: '2020-09-01T10:45:02.000Z', target: '2020-09-01T10:45:00.000Z' },
        { value: 'zzz', target: '2020-09-01T10:45:00.000Z' },
        { value: '', target: '2020-09-01T10:45:00.000Z' },
        { value: undefined, target: '2020-09-01T10:45:00.000Z' },
        { value: null, target: '2020-09-01T10:45:00.000Z' },
        { value: NaN, target: '2020-09-01T10:45:00.000Z' },
        { value: 5, target: '2020-09-01T10:45:00.000Z' },
      ];

      examples.forEach(({ value, target }) => {
        test(`${value} is not near ${target}`, () => {
          expect(value).not.toBeIsoDateStringNear(target);
        });
      });

      examples.forEach(({ value, target }) => {
        test(`failure message: ${value} near ${target}`, () => {
          try {
            expect(value).toBeIsoDateStringNear(target);
          } catch(e) {
            expect(e.message).toMatchSnapshot();
          }
        });
      });
    });

  });

  describe('custom threshold', () => {

    describe('affirmative', () => {

      const examples = [
        { value: '2020-09-01T10:45:00.010Z', target: '2020-09-01T10:45:00.000Z', threshold: 10, units: 'milliseconds' },
        { value: '2020-09-01T10:44:59.990Z', target: '2020-09-01T10:45:00.000Z', threshold: 10, units: 'milliseconds' },

        { value: '2020-09-01T10:45:10.000Z', target: '2020-09-01T10:45:00.000Z', threshold: 10, units: 'seconds' },
        { value: '2020-09-01T10:44:50.000Z', target: '2020-09-01T10:45:00.000Z', threshold: 10, units: 'seconds' },

        { value: '2020-09-01T10:55:00.000Z', target: '2020-09-01T10:45:00.000Z', threshold: 10, units: 'minutes' },
        { value: '2020-09-01T10:35:00.000Z', target: '2020-09-01T10:45:00.000Z', threshold: 10, units: 'minutes' },

        { value: '2020-09-01T20:45:00.000Z', target: '2020-09-01T10:45:00.000Z', threshold: 10, units: 'hours' },
        { value: '2020-09-01T00:45:00.000Z', target: '2020-09-01T10:45:00.000Z', threshold: 10, units: 'hours' },

        { value: '2020-09-25T10:45:00.000Z', target: '2020-09-15T10:45:00.000Z', threshold: 10, units: 'days' },
        { value: '2020-09-05T10:45:00.000Z', target: '2020-09-15T10:45:00.000Z', threshold: 10, units: 'days' },
      ];

      examples.forEach(({ value, target, threshold, units }) => {
        test(`${value} is within ${threshold} ${units} of ${target}`, () => {
          expect(value).toBeIsoDateStringNear(target, { threshold, units });
        });
      });

      examples.forEach(({ value, target, threshold, units }) => {
        test(`failure message: ${value} within ${threshold} ${units} of ${target}`, () => {
          try {
            expect(value).not.toBeIsoDateStringNear(target, { threshold, units });
          } catch(e) {
            expect(e.message).toMatchSnapshot();
          }
        });
      });

    });

    describe('negative', () => {

      const examples = [
        { value: '2020-09-01T10:45:00.020Z', target: '2020-09-01T10:45:00.000Z', threshold: 10, units: 'milliseconds' },
        { value: 'zzz', target: '2020-09-01T10:45:00.000Z', threshold: 10, units: 'milliseconds' },
        { value: 5, target: '2020-09-01T10:45:00.000Z', threshold: 10, units: 'milliseconds' },
        { value: NaN, target: '2020-09-01T10:45:00.000Z', threshold: 10, units: 'milliseconds' },
        { value: undefined, target: '2020-09-01T10:45:00.000Z', threshold: 10, units: 'milliseconds' },
        { value: null, target: '2020-09-01T10:45:00.000Z', threshold: 10, units: 'milliseconds' },
        { value: '', target: '2020-09-01T10:45:00.000Z', threshold: 10, units: 'milliseconds' },

        { value: '2020-09-01T10:45:00.020Z', target: '2020-09-01T10:45:00.000Z', threshold: 10, units: 'milliseconds' },
        { value: '2020-09-01T10:44:59.980Z', target: '2020-09-01T10:45:00.000Z', threshold: 10, units: 'milliseconds' },

        { value: '2020-09-01T10:45:20.000Z', target: '2020-09-01T10:45:00.000Z', threshold: 10, units: 'seconds' },
        { value: '2020-09-01T10:44:40.000Z', target: '2020-09-01T10:45:00.000Z', threshold: 10, units: 'seconds' },

        { value: '2020-09-01T10:56:00.000Z', target: '2020-09-01T10:45:00.000Z', threshold: 10, units: 'minutes' },
        { value: '2020-09-01T10:34:00.000Z', target: '2020-09-01T10:45:00.000Z', threshold: 10, units: 'minutes' },

        { value: '2020-09-01T21:45:00.000Z', target: '2020-09-01T10:45:00.000Z', threshold: 10, units: 'hours' },
        { value: '2020-08-30T23:45:00.000Z', target: '2020-09-01T10:45:00.000Z', threshold: 10, units: 'hours' },

        { value: '2020-09-26T10:45:00.000Z', target: '2020-09-15T10:45:00.000Z', threshold: 10, units: 'days' },
        { value: '2020-09-04T10:45:00.000Z', target: '2020-09-15T10:45:00.000Z', threshold: 10, units: 'days' },
      ];

      examples.forEach(({ value, target, threshold, units }) => {
        test(`${value} is not within ${threshold} ${units} of ${target}`, () => {
          expect(value).not.toBeIsoDateStringNear(target, { threshold, units });
        });
      });

      examples.forEach(({ value, target, threshold, units }) => {
        test(`failure message: ${value} within ${threshold} ${units} of ${target}`, () => {
          try {
            expect(value).toBeIsoDateStringNear(target, { threshold, units });
          } catch(e) {
            expect(e.message).toMatchSnapshot();
          }
        });
      });

    });

  });

});
