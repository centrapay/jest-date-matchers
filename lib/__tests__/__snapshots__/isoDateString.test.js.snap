// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`toBeIsoDateStringNear custom threshold affirmative failure message: 2020-09-01T00:45:00.000Z within 10 hours of 2020-09-01T10:45:00.000Z 1`] = `"expected 2020-09-01T00:45:00.000Z not to be ISO Date String within 10 hours of 2020-09-01T10:45:00.000Z (was 10 hours)"`;

exports[`toBeIsoDateStringNear custom threshold affirmative failure message: 2020-09-01T10:35:00.000Z within 10 minutes of 2020-09-01T10:45:00.000Z 1`] = `"expected 2020-09-01T10:35:00.000Z not to be ISO Date String within 10 minutes of 2020-09-01T10:45:00.000Z (was 10 minutes)"`;

exports[`toBeIsoDateStringNear custom threshold affirmative failure message: 2020-09-01T10:44:50.000Z within 10 seconds of 2020-09-01T10:45:00.000Z 1`] = `"expected 2020-09-01T10:44:50.000Z not to be ISO Date String within 10 seconds of 2020-09-01T10:45:00.000Z (was 10 seconds)"`;

exports[`toBeIsoDateStringNear custom threshold affirmative failure message: 2020-09-01T10:44:59.990Z within 10 milliseconds of 2020-09-01T10:45:00.000Z 1`] = `"expected 2020-09-01T10:44:59.990Z not to be ISO Date String within 10 milliseconds of 2020-09-01T10:45:00.000Z (was 10 milliseconds)"`;

exports[`toBeIsoDateStringNear custom threshold affirmative failure message: 2020-09-01T10:45:00.010Z within 10 milliseconds of 2020-09-01T10:45:00.000Z 1`] = `"expected 2020-09-01T10:45:00.010Z not to be ISO Date String within 10 milliseconds of 2020-09-01T10:45:00.000Z (was 10 milliseconds)"`;

exports[`toBeIsoDateStringNear custom threshold affirmative failure message: 2020-09-01T10:45:10.000Z within 10 seconds of 2020-09-01T10:45:00.000Z 1`] = `"expected 2020-09-01T10:45:10.000Z not to be ISO Date String within 10 seconds of 2020-09-01T10:45:00.000Z (was 10 seconds)"`;

exports[`toBeIsoDateStringNear custom threshold affirmative failure message: 2020-09-01T10:55:00.000Z within 10 minutes of 2020-09-01T10:45:00.000Z 1`] = `"expected 2020-09-01T10:55:00.000Z not to be ISO Date String within 10 minutes of 2020-09-01T10:45:00.000Z (was 10 minutes)"`;

exports[`toBeIsoDateStringNear custom threshold affirmative failure message: 2020-09-01T20:45:00.000Z within 10 hours of 2020-09-01T10:45:00.000Z 1`] = `"expected 2020-09-01T20:45:00.000Z not to be ISO Date String within 10 hours of 2020-09-01T10:45:00.000Z (was 10 hours)"`;

exports[`toBeIsoDateStringNear custom threshold affirmative failure message: 2020-09-05T10:45:00.000Z within 10 days of 2020-09-15T10:45:00.000Z 1`] = `"expected 2020-09-05T10:45:00.000Z not to be ISO Date String within 10 days of 2020-09-15T10:45:00.000Z (was 10 days)"`;

exports[`toBeIsoDateStringNear custom threshold affirmative failure message: 2020-09-25T10:45:00.000Z within 10 days of 2020-09-15T10:45:00.000Z 1`] = `"expected 2020-09-25T10:45:00.000Z not to be ISO Date String within 10 days of 2020-09-15T10:45:00.000Z (was 10 days)"`;

exports[`toBeIsoDateStringNear custom threshold negative failure message:  within 10 milliseconds of 2020-09-01T10:45:00.000Z 1`] = `"expected  to be ISO Date String within 10 milliseconds of 2020-09-01T10:45:00.000Z (was not an ISO date string)"`;

exports[`toBeIsoDateStringNear custom threshold negative failure message: 5 within 10 milliseconds of 2020-09-01T10:45:00.000Z 1`] = `"expected 5 to be ISO Date String within 10 milliseconds of 2020-09-01T10:45:00.000Z (was not a string)"`;

exports[`toBeIsoDateStringNear custom threshold negative failure message: 2020-08-30T23:45:00.000Z within 10 hours of 2020-09-01T10:45:00.000Z 1`] = `"expected 2020-08-30T23:45:00.000Z to be ISO Date String within 10 hours of 2020-09-01T10:45:00.000Z (was 35 hours)"`;

exports[`toBeIsoDateStringNear custom threshold negative failure message: 2020-09-01T10:34:00.000Z within 10 minutes of 2020-09-01T10:45:00.000Z 1`] = `"expected 2020-09-01T10:34:00.000Z to be ISO Date String within 10 minutes of 2020-09-01T10:45:00.000Z (was 11 minutes)"`;

exports[`toBeIsoDateStringNear custom threshold negative failure message: 2020-09-01T10:44:40.000Z within 10 seconds of 2020-09-01T10:45:00.000Z 1`] = `"expected 2020-09-01T10:44:40.000Z to be ISO Date String within 10 seconds of 2020-09-01T10:45:00.000Z (was 20 seconds)"`;

exports[`toBeIsoDateStringNear custom threshold negative failure message: 2020-09-01T10:44:59.980Z within 10 milliseconds of 2020-09-01T10:45:00.000Z 1`] = `"expected 2020-09-01T10:44:59.980Z to be ISO Date String within 10 milliseconds of 2020-09-01T10:45:00.000Z (was 20 milliseconds)"`;

exports[`toBeIsoDateStringNear custom threshold negative failure message: 2020-09-01T10:45:00.020Z within 10 milliseconds of 2020-09-01T10:45:00.000Z 1`] = `"expected 2020-09-01T10:45:00.020Z to be ISO Date String within 10 milliseconds of 2020-09-01T10:45:00.000Z (was 20 milliseconds)"`;

exports[`toBeIsoDateStringNear custom threshold negative failure message: 2020-09-01T10:45:00.020Z within 10 milliseconds of 2020-09-01T10:45:00.000Z 2`] = `"expected 2020-09-01T10:45:00.020Z to be ISO Date String within 10 milliseconds of 2020-09-01T10:45:00.000Z (was 20 milliseconds)"`;

exports[`toBeIsoDateStringNear custom threshold negative failure message: 2020-09-01T10:45:20.000Z within 10 seconds of 2020-09-01T10:45:00.000Z 1`] = `"expected 2020-09-01T10:45:20.000Z to be ISO Date String within 10 seconds of 2020-09-01T10:45:00.000Z (was 20 seconds)"`;

exports[`toBeIsoDateStringNear custom threshold negative failure message: 2020-09-01T10:56:00.000Z within 10 minutes of 2020-09-01T10:45:00.000Z 1`] = `"expected 2020-09-01T10:56:00.000Z to be ISO Date String within 10 minutes of 2020-09-01T10:45:00.000Z (was 11 minutes)"`;

exports[`toBeIsoDateStringNear custom threshold negative failure message: 2020-09-01T21:45:00.000Z within 10 hours of 2020-09-01T10:45:00.000Z 1`] = `"expected 2020-09-01T21:45:00.000Z to be ISO Date String within 10 hours of 2020-09-01T10:45:00.000Z (was 11 hours)"`;

exports[`toBeIsoDateStringNear custom threshold negative failure message: 2020-09-04T10:45:00.000Z within 10 days of 2020-09-15T10:45:00.000Z 1`] = `"expected 2020-09-04T10:45:00.000Z to be ISO Date String within 10 days of 2020-09-15T10:45:00.000Z (was 11 days)"`;

exports[`toBeIsoDateStringNear custom threshold negative failure message: 2020-09-26T10:45:00.000Z within 10 days of 2020-09-15T10:45:00.000Z 1`] = `"expected 2020-09-26T10:45:00.000Z to be ISO Date String within 10 days of 2020-09-15T10:45:00.000Z (was 11 days)"`;

exports[`toBeIsoDateStringNear custom threshold negative failure message: NaN within 10 milliseconds of 2020-09-01T10:45:00.000Z 1`] = `"expected NaN to be ISO Date String within 10 milliseconds of 2020-09-01T10:45:00.000Z (was not a string)"`;

exports[`toBeIsoDateStringNear custom threshold negative failure message: null within 10 milliseconds of 2020-09-01T10:45:00.000Z 1`] = `"expected null to be ISO Date String within 10 milliseconds of 2020-09-01T10:45:00.000Z (was not a string)"`;

exports[`toBeIsoDateStringNear custom threshold negative failure message: undefined within 10 milliseconds of 2020-09-01T10:45:00.000Z 1`] = `"expected undefined to be ISO Date String within 10 milliseconds of 2020-09-01T10:45:00.000Z (was not a string)"`;

exports[`toBeIsoDateStringNear custom threshold negative failure message: zzz within 10 milliseconds of 2020-09-01T10:45:00.000Z 1`] = `"expected zzz to be ISO Date String within 10 milliseconds of 2020-09-01T10:45:00.000Z (was not an ISO date string)"`;

exports[`toBeIsoDateStringNear default threshold affirmative failure message: 2020-09-01T10:44:59.000Z near 2020-09-01T10:45:00.000Z 1`] = `"expected 2020-09-01T10:44:59.000Z not to be ISO Date String within 1 seconds of 2020-09-01T10:45:00.000Z (was 1 seconds)"`;

exports[`toBeIsoDateStringNear default threshold affirmative failure message: 2020-09-01T10:45:00.000Z near 2020-09-01T10:45:00.000Z 1`] = `"expected 2020-09-01T10:45:00.000Z not to be ISO Date String within 1 seconds of 2020-09-01T10:45:00.000Z (was 0 seconds)"`;

exports[`toBeIsoDateStringNear default threshold affirmative failure message: 2020-09-01T10:45:01.000Z near 2020-09-01T10:45:00.000Z 1`] = `"expected 2020-09-01T10:45:01.000Z not to be ISO Date String within 1 seconds of 2020-09-01T10:45:00.000Z (was 1 seconds)"`;

exports[`toBeIsoDateStringNear default threshold negative failure message:  near 2020-09-01T10:45:00.000Z 1`] = `"expected  to be ISO Date String within 1 seconds of 2020-09-01T10:45:00.000Z (was not an ISO date string)"`;

exports[`toBeIsoDateStringNear default threshold negative failure message: 5 near 2020-09-01T10:45:00.000Z 1`] = `"expected 5 to be ISO Date String within 1 seconds of 2020-09-01T10:45:00.000Z (was not a string)"`;

exports[`toBeIsoDateStringNear default threshold negative failure message: 2020-09-01T10:44:58.000Z near 2020-09-01T10:45:00.000Z 1`] = `"expected 2020-09-01T10:44:58.000Z to be ISO Date String within 1 seconds of 2020-09-01T10:45:00.000Z (was 2 seconds)"`;

exports[`toBeIsoDateStringNear default threshold negative failure message: 2020-09-01T10:45:02.000Z near 2020-09-01T10:45:00.000Z 1`] = `"expected 2020-09-01T10:45:02.000Z to be ISO Date String within 1 seconds of 2020-09-01T10:45:00.000Z (was 2 seconds)"`;

exports[`toBeIsoDateStringNear default threshold negative failure message: NaN near 2020-09-01T10:45:00.000Z 1`] = `"expected NaN to be ISO Date String within 1 seconds of 2020-09-01T10:45:00.000Z (was not a string)"`;

exports[`toBeIsoDateStringNear default threshold negative failure message: null near 2020-09-01T10:45:00.000Z 1`] = `"expected null to be ISO Date String within 1 seconds of 2020-09-01T10:45:00.000Z (was not a string)"`;

exports[`toBeIsoDateStringNear default threshold negative failure message: undefined near 2020-09-01T10:45:00.000Z 1`] = `"expected undefined to be ISO Date String within 1 seconds of 2020-09-01T10:45:00.000Z (was not a string)"`;

exports[`toBeIsoDateStringNear default threshold negative failure message: zzz near 2020-09-01T10:45:00.000Z 1`] = `"expected zzz to be ISO Date String within 1 seconds of 2020-09-01T10:45:00.000Z (was not an ISO date string)"`;
